File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/WebSockets/DynamicData/DynamicDataForm.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;

using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;
using System.Threading;
using Gizmox.WebGUI.Common.WebSockets;
using Newtonsoft.Json;

#endregion

namespace WebSocketsSampleAppsCS
{
    public partial class DynamicDataForm : Form, IPageHandler
    {

        public DynamicDataForm()
        {
            //Sets WebSocket's handler class
            Application.WebSockets.SetHandler<DynamicDataWebSocketsHandler>();
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Load event of the DynamicDataPage control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void DynamicDataPage_Load(object sender, EventArgs e)
        {
            //Tries to cast WebSocketHandler as CompanionKitWebSocketsHandler
            DynamicDataWebSocketsHandler objHandler = Application.WebSockets.WebSocketHandler as DynamicDataWebSocketsHandler;
            if (objHandler != null)
            {
                //Sets current instance to Page property
                objHandler.Page = this;
            }
        }

        /// <summary>
        /// Pages the handle.
        /// </summary>
        /// <param name="strConnectionId">The STR connection id.</param>
        /// <param name="strData">The STR data.</param>
        /// <exception cref="System.NotImplementedException"></exception>
        public void PageHandle(string strConnectionId, string strData)
        {
            //Not implemented. Retained for interface compatibility. 
            throw new NotImplementedException();
        }

        /// <summary>
        /// Pages the client handle.
        /// </summary>
        /// <param name="objSender">The obj sender.</param>
        /// <param name="objArgs">The <see cref="ClientEventArgs" /> instance containing the event data.</param>
        public void PageClientHandle(object objSender, Gizmox.WebGUI.Forms.Client.ClientEventArgs objArgs)
        {
            //Invokes js ClientHandler function
            objArgs.Context.Invoke("ClientHandler");
        }

       

       /// <summary>
       /// Called when [unregister components].
       /// </summary>
       protected override void OnUnregisterComponents()
       {
           base.OnUnregisterComponents();
           ((DynamicDataWebSocketsHandler)(Application.WebSockets.WebSocketHandler)).Abort();
       }

    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Applications/WebSockets/WebSocketsSampleApps/VB/DynamicDataForm.vb
Imports Gizmox.WebGUI.Forms
Imports Gizmox.WebGUI.Common

Namespace Gizmox.WebGUI.Forms.CompanionKit.WebSocketsSampleAppsVB
    Partial Public Class DynamicDataForm
        Inherits Form
        Implements IPageHandler

        Public Sub New()
            'Sets WebSocket's handler class
            Application.WebSockets.SetHandler(Of DynamicDataWebSocketsHandler)()
            InitializeComponent()
        End Sub

        ''' <summary>
        ''' Handles the Load event of the DynamicDataPage control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub DynamicDataPage_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            'Tries to cast WebSocketHandler as CompanionKitWebSocketsHandler
            Dim objHandler As DynamicDataWebSocketsHandler = TryCast(Application.WebSockets.WebSocketHandler, DynamicDataWebSocketsHandler)
            If objHandler IsNot Nothing Then
                'Sets current instance to Page property
                objHandler.Page = Me
            End If
        End Sub

        ''' <summary>
        ''' Pages the handle.
        ''' </summary>
        ''' <param name="strConnectionId">The STR connection id.</param>
        ''' <param name="strData">The STR data.</param>
        ''' <exception cref="System.NotImplementedException"></exception>
        Public Sub PageHandle(ByVal strConnectionId As String, ByVal strData As String) Implements IPageHandler.PageHandle
            'Not implemented. Retained for interface compatibility. 
            Throw New NotImplementedException()
        End Sub

        ''' <summary>
        ''' Pages the client handle.
        ''' </summary>
        ''' <param name="objSender">The obj sender.</param>
        ''' <param name="objArgs">The <see cref="ClientEventArgs" /> instance containing the event data.</param>
        Public Sub PageClientHandle(ByVal objSender As Object, ByVal objArgs As Client.ClientEventArgs) Implements IPageHandler.PageClientHandle
            'Invokes js ClientHandler function
            objArgs.Context.Invoke("ClientHandler")
        End Sub



        ''' <summary>
        ''' Called when [unregister components].
        ''' </summary>
        Protected Overrides Sub OnUnregisterComponents()
            MyBase.OnUnregisterComponents()
            DirectCast(Application.WebSockets.WebSocketHandler, DynamicDataWebSocketsHandler).Abort()
        End Sub

    End Class
End Namespace